notes.go

```diff
@@ -0,0 +1,97 @@
+package chartmogul
+
+// Note is the customer note as represented in the API.
+type Note struct {
+	UUID string `json:"uuid"`
+	// Basic info
+	CustomerUUID string `json:"customer_uuid"`
+	Type         string `json:"type"`
+	Text         string `json:"text"`
+	Author       string `json:"author"`
+	CallDuration uint32 `json:"call_duration"`
+	CreatedAt    string `json:"created_at"`
+	UpdatedAt    string `json:"updated_at"`
+}
+
+// UpdateNote allows updating note on the update endpoint.
+type UpdateNote struct {
+	Text         string `json:"text"`
+	AuthorEmail  string `json:"author_email"`
+	CallDuration uint32 `json:"call_duration"`
+	CreatedAt    string `json:"create_at"`
+	UpdatedAt    string `json:"updated_at"`
```

Comment:
We probably don't want to send these timestamps in the request

Reply:
We have a spec for sending `updated_at` timestamp for updating a note

Reply:
Ah I see. This seems like a bad idea, timestamps like these should be auto-generated or at least have a consistent meaning. I presume the reason behind it is there is a need for a timestamp field for something beyond the creation/update of the record in which case it should be a separate column(s). 

Can you check to see if this is really the desired behavior?

Reply:
@daesu The reason is for backfilling notes. Users specifically wanted to have a specific date when they're backfilling. See https://www.notion.so/API-endpoints-for-adding-notes-call-logs-798b788a08764549a334db5d48d65469?d=146bd3b41feb4d4ba436aa8720e184c8&pvs=4#df1e2a9458504230b4101a842d9eccb2 and https://chartmogul.slack.com/archives/C04R1AV2HU4/p1701270734531819.

Reply:
@briwa I see the requirement from Katarina. Shouldn't we then add something like an `external_created_at` (and `external_updated_at`) column to handle the need to set from the API? 

If we allow `created_at` to be set from the API then it loses it's value in terms of auditing, data consistency, and will lead to confusion. What does `created_at` mean for a record when it's not referring to the timestamp the record was created at? How do we know if it was set by the API or generated automatically?

Reply:
@daesu I see what you mean. 

>  it loses it's value in terms of auditing, data consistency, and will lead to confusion

In other cases, you might be right, but I'm not too sure if it would lose its value or affect auditing or data consistency too much. Adding two new columns only to identify if the update comes from the API is overkill, IMO. We use the same column (`created_at`, `updated_at`) to list the customer notes and also to show them in the UI as a timeline, so if anything, adding the new columns would just simply add unnecessary complexity. If it's just to know if the records have been ever updated by the API, we could either see the log (the endpoint URL is already different than the one in the UI) or we could check our own reporting system which we are flagging it already ([example for creating a note](https://github.com/chartmogul/platform/blob/e022469cd5e088cf38dbdab767c85e51a2c37022/app/api/endpoints/v1/helpers/customer_notes.rb#L96)).

But this is just my take. Feel free to raise it further if you think this is a security concern, but in my opinion, for this context only, having the same column name and params to be updated for a clear reasoning isn't too much of a harm

Reply:
So I'm thinking about changing the column name, as Paul suggested.

As a payload,
```
type,
text,
note_created_at,
note_updated_at,
....
```

For response,
```
type,
text,
created_at,
update_at
```

But if we need to think about consistency, or if you have different ideas, please feel free to let me know!

Reply:
@SoeunSona

> changing the column name

if by changing column names you mean just renaming the params, I don't think it addresses what Paul said. His main concern is not being able to determine if the values in the `created_at` and `updated_at` columns are updated from the API or somewhere else. I think what he meant is to actually add two more columns in the DB

Reply:
@SoeunSona

> changing the column name

if by changing column names you mean just renaming the params, I don't think it addresses what Paul said. His main concern is not being able to determine if the values in the `created_at` and `updated_at` columns are updated from the API or somewhere else. Renaming the params to something else won't solve that since in the end, it still updates the same `created_at` and `updated_at` columns. I think what he meant is to actually add two more different columns in the DB reserved for external updates

Reply:
@SoeunSona @daesu 

> changing the column name

I'm still not too sure about renaming the column. In our system (Rails), `created_at` and `updated_at` is pretty much the standard to know if a record is created and updated. The references are everywhere so changing the name, while it works the same and not too different (`created_at` -> `note_created_at`), still feels unnatural to me, and to change it just to cater to this edge case (backfilling) is actually a lot to change.

I saw [Sona's branch](https://github.com/chartmogul/platform/compare/main...soeunlee/rename_created_at_updated_at) on changing the column but that isn't just it -- we need to change the references on the response serializer in the UI, how it's sorted in the customer timeline query response, how it's sorted in the API, how it's being used in the client app, etc...

As I mentioned above, ideally in practice it's not a bad idea and if we wanted to be strict, if customer notes are something that would have an impact on data accuracy or included in the reports, we should probably do something different indeed. But in this context, I still think that updating the columns directly is harmless enough for a note related to a customer. 

I'd advise to stick to the original spec and see how the users use it first. Or we could discuss it in a bigger setting instead of here. This is a public repo, anyway. Something like this would probably have more benefit to be discussed internally